staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -46,18 +52,14 @@ type ApplySetParentRef struct {
 }
 
 // NewApplySet creates a new ApplySet object from a parent reference in the format [RESOURCE][.GROUP]/NAME
-func NewApplySet(parentRefStr string, namespace string, mapper meta.RESTMapper) (*ApplySet, error) {
-	var parent *ApplySetParentRef
-	var err error
-
-	if parent, err = parentRefFromStr(parentRefStr, mapper); err != nil {
+func NewApplySet(parentRefStr string, nsFromFlag string, mapper meta.RESTMapper) (*ApplySet, error) {
```

Comment:
At some stage we might want two separate methods - one which deals with the flags, and one is explicit.  The more we can avoid pushing knowledge of flags down deep into the logic, the simpler it will be IMO.  That said, no need for the second method yet!